docs/source/fx.rst

```diff
@@ -10,7 +10,261 @@ Overview
+        if dim0 == 3:
+            return torch.relu(x)
+        else:
+            return torch.neg(x)
+
+    traced = torch.fx.symbolic_trace(func_to_trace)
+    """
+      <...>
+      File "dyn.py", line 6, in func_to_trace
+        if dim0 == 3:
+      File "pytorch/torch/fx/proxy.py", line 155, in __bool__
+        return self.tracer.to_bool(self)
+      File "pytorch/torch/fx/proxy.py", line 85, in to_bool
+        raise TraceError('symbolically traced variables cannot be used as inputs to control flow')
+    torch.fx.proxy.TraceError: symbolically traced variables cannot be used as inputs to control flow
+    """
+
+The condition to the ``if`` statement relies on the value of ``dim0``,
+which eventually relies on the value of ``x``, a function input. Since
+``x`` can change (i.e. if you pass a new input tensor to the traced
+function), this is *dynamic control flow*. The traceback walks back up
+through your code to show you where this situation happens.
+
+Static Control Flow
+~~~~~~~~~~~~~~~~~~~
+
+On the other hand, so called *static control flow* is supported. Static
+control flow is loops or ``if`` statements whose value cannot change
+across invocations. Typically, in PyTorch programs, this control flow
+arises for code making decisions about a model’s architecture based on
+hyper-parameters. As a concrete example:
+
+::
+
+    import torch
+    import torch.fx
+
+    class MyModule(torch.nn.Module):
+        def __init__(self, do_activation : bool = False):
+            super().__init__()
+            self.do_activation = do_activation
+            self.linear = torch.nn.Linear(512, 512)
+
+        def forward(self, x):
+            x = self.linear(x)
+            # This if-statement is so-called static control flow.
+            # Its condition does not depend on any input values
+            if self.do_activation:
+                x = torch.relu(x)
+            return x
+
+    without_activation = MyModule(do_activation=False)
+    with_activation = MyModule(do_activation=True)
+
+    traced_without_activation = torch.fx.symbolic_trace(without_activation)
+    print(traced_without_activation.code)
+    """
+    def forward(self, x):
+        linear_1 = self.linear(x);  x = None
+        return linear_1
+    """
+
+    traced_with_activation = torch.fx.symbolic_trace(with_activation)
+    print(traced_with_activation.code)
+    """
+    import torch
+    def forward(self, x):
+        linear_1 = self.linear(x);  x = None
+        relu_1 = torch.relu(linear_1);  linear_1 = None
+        return relu_1
+    """
+
+The if-statement ``if self.do_activation`` does not depend on any
+function inputs, thus it is static. ``do_activation`` can be considered
+to be a hyper-parameter, and the traces of different instances of
+``MyModule`` with different values for that parameter have different
+code. This is a valid pattern that is supported by symbolic tracing.
+
+Many instances of dynamic control flow are semantically static control
```

Comment:
Might be worth adding something about the workaround we mentioned previously

Ie:

```
def f(x, flag):
    if flag: return x
    else: return x*2
    
fx.symbolic_trace(f) # Fails!

def g(flag):
    return lambda x: f(x, flag)
    
new_f = g(flag=True)
fx.symbolic_trace(new_f)
```